src/main/java/net/snowflake/client/jdbc/SFConnectionHandler.java

```diff
@@ -50,4 +50,6 @@ SnowflakeBaseResultSet createAsyncResultSet(SFBaseResultSet resultSet, Statement
    */
   SFBaseFileTransferAgent getFileTransferAgent(String command, SFBaseStatement statement)
       throws SQLNonTransientConnectionException, SnowflakeSQLException;
+
+  public abstract String getBindStageName();
```

Comment:
Methods in interface are public abstract by default, so not need to explicitly add that.
Snowpark client has its own implementation of the connection handler, so we need to update that implementation as well once this checks in.

Reply:
It might be a breaking change for customers that implements their own connection handler as well. Is this a concern?

Reply:
Yeah I'm wondering if we should have a default value for this given it adds a new method in interface.

Reply:
I am unclear why we have this API here at all. The connectionHandler should not have to know the bindStageName.

If this was done to have the default "SYSTEM$BIND" stage name preserved for the jdbc case, then why not have this logic in SFBaseSession. It can just use the default value unless the stageName is set via `setArrayBindStage`.

Reply:
The idea is that the client mode will have a name, `SYSTEM$BIND`, and that we will have a different one for stored procs, that can be overridden by the `ConnectionHandler`.

`setArrayBindStage` is actually a bit of a misnomer. In JDBC, it's used as a "flag" of sorts to indicate that the bind stage has been created and determined for the session.

I thought I would include it in the connection handler because it's not going to be different from session to session. I can definitely make this has a default value, though.